l1-contracts/contracts/governance/ChainAdmin.sol

```diff
@@ -4,13 +4,25 @@ pragma solidity 0.8.24;
 
 import {Ownable2Step} from "@openzeppelin/contracts/access/Ownable2Step.sol";
 import {IChainAdmin} from "./IChainAdmin.sol";
+import {IRestriction} from "./IRestriction.sol";
+import { Call } from "./Common.sol";
+
+import {EnumerableSet} from "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";
+import {SafeCast} from "@openzeppelin/contracts/utils/math/SafeCast.sol";
 
 /// @author Matter Labs
 /// @custom:security-contact security@matterlabs.dev
 /// @notice The contract is designed to hold the `admin` role in ZKSync Chain (State Transition) contracts.
 /// The owner of the contract can perform any external calls and also save the information needed for
 /// the blockchain node to accept the protocol upgrade.
 contract ChainAdmin is IChainAdmin, Ownable2Step {
```

Comment:
Please note, the contract is `Ownable2Step` so it can transfer ownership without restriction checks :(

Reply:
It should not matter, since the `admin` of the chain is still this contract. All actions of the owner are proxied through this contract